### PR TITLE
Added extra percentile memory metrics for the ASP.NET Benchmarks

### PR DESCRIPTION
**Pull request by mrsharm**
This PR gets the obtains the following metrics + interprets them in the notebooks

- Working Set P99 (MB) 
- Working Set P95 (MB)
- Working Set P90 (MB)
- Working Set P75 (MB)
- Working Set P50 (MB)
- Private Memory P99 (MB)
- Private Memory P95 (MB)
- Private Memory P90 (MB)
- Private Memory P75 (MB)
- Private Memory P50 (MB)

Example output from crank of pertinent rows:

| application                             |                                 |                                           |
| --------------------------------------- | ------------------------------- | ----------------------------------------- |
| Max Working Set (MB)                    | 516                             | |
| Max Private Memory (MB)                 | 544                             | |
| P99 Working Set (MB)                    | 516                             |                                           |
| P95 Working Set (MB)                    | 514                             |                                           |
| P90 Working Set (MB)                    | 512                             |                                           |
| P75 Working Set (MB)                    | 509                             |                                           |
| P50 Working Set (MB)                    | 507                             |                                           |
| P99 Private Memory (MB)                 | 544                             |                                           |
| P95 Private Memory (MB)                 | 541                             |                                           |
| P90 Private Memory (MB)                 | 541                             |                                           |
| P75 Private Memory (MB)                 | 537                             |                                           |
| P50 Private Memory (MB)                 | 536                             |                                           |

### CODE REVIEW

File: src/benchmarks/gc/GC.Infrastructure/GC.Infrastructure/Commands/RunCommand/BaseSuite/PercentileBasedMetricsConfiguration.yml [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+
+- name: benchmarks/private-memory/P50
+  description: Private Memory P50 (MB)
+  format: n0
+
+onResultsCreating:
+  - |
+
+    function steadyState(measurement, reduce) {
+      // Calculates the result of the 'reduce' operation on the 'measurement' property of the application job.
```
Comment (mrsharm) [inline src/benchmarks/gc/GC.Infrastructure/GC.Infrastructure/Commands/RunCommand/BaseSuite/PercentileBasedMetricsConfiguration.yml:49]: Change name to applyPercentile.

**Review by Maoni0 — DISMISSED** [review]
LGTM nit the name change above

**Review by cshung — APPROVED** [review]
LGTM

**Comment by mrsharm** [comment]
@cincuranet, could we please merge this PR?

**Comment by cincuranet** [comment]
@mrsharm Done.

**Pull request merged**